bam/add_example_test.go

```diff
@@ -22,7 +23,7 @@ func ExampleIndex_Add() {
 	var bai bam.Index
 	for {
 		r, err := br.Read()
-		if err == io.EOF {
+		if errors.Is(err, io.EOF) {
```

Comment:
`io.EOF` is never wrapped, so I don't think these are necessary.

Reply:
I kind of agree that `io.EOF` should never be wrapped, but I've encountered readers where it happens. Anyways, I can revert this change.

Reply:
Note, it's usually due to some oversight in a wrapping reader writer that propagates some internal error across some misc. logic.